pesticide/test_helpers.go

```diff
@@ -0,0 +1,99 @@
+/**
+ * Pesticide are a package to help you to test your functions on a template
+ * engine. It provides a set of test cases that you can use to test your
+ * functions.
+ * More pesticide for less bugs.
+ */
+package pesticide
+
+import (
+	"bytes"
+	"testing"
+	"text/template"
+
+	"github.com/go-sprout/sprout"
+	"github.com/go-sprout/sprout/registry/maps"
+	"github.com/go-sprout/sprout/registry/reflect"
+	"github.com/go-sprout/sprout/registry/slices"
+	"github.com/go-sprout/sprout/registry/strings"
+	"github.com/stretchr/testify/assert"
+)
+
+type TestCase struct {
+	Name     string
+	Input    string
+	Expected string
+	Data     map[string]any
+}
+
+type MustTestCase struct {
+	TestCase
+	ExpectedErr string
+}
+
+func RunTestCases(t *testing.T, registry sprout.Registry, tc []TestCase) {
+	t.Helper()
+
+	for _, test := range tc {
+		t.Run(test.Name, func(t *testing.T) {
+			t.Helper()
+
+			tmplResponse, err := runTemplate(t, testHandler(registry), test.Input, test.Data)
+			assert.NoError(t, err)
+			assert.Equal(t, test.Expected, tmplResponse)
+		})
+	}
+}
+
+func RunMustTestCases(t *testing.T, registry sprout.Registry, tc []MustTestCase) {
+	t.Helper()
+
+	for _, test := range tc {
+		t.Run(test.Name, func(t *testing.T) {
+			t.Helper()
+
+			tmplResponse, err := runTemplate(t, testHandler(registry), test.Input, test.Data)
+			if test.ExpectedErr != "" {
+				if assert.Error(t, err) {
+					assert.Contains(t, err.Error(), test.ExpectedErr)
+				}
```

Comment:
This can be replaced by


```suggestion
				assert.ErrorContains(t, err, test.ExpectedErr)
```